### ISSUE TITLE
Koch bootCI error on FreeBSD

### ISSUE THREAD

**Issue by euantorano** [nim-lang/Nim#12113]
When running `koch runCI` on FreeBSD, the following error is reported:

```
[build@build /usr/home/build/Nim]$ ./koch runCI
runCI:
'/usr/home/build/Nim/koch' boot
sh: Syntax error: Unterminated quoted string
FAILURE
```

When running `echo $SHELL`, the shell is `/usr/local/bin/bash`.

Running `'/usr/home/build/Nim/koch' boot` from the shell works correctly.

**Comment by euantorano** [comment]
This may be related to [`quoteShellPosix`](https://github.com/nim-lang/Nim/blob/296dfae8af61a9ce145affec791ef338b3774950/lib/pure/os.nim#L864) - reading that code, `/usr/home/build/Nim/koch` shouldn't be quoted at all as all of the characters in that path are in the set of safe characters aren't they?

**Comment by awr1** [comment]
We should detect which shell is the default shell when running CI (I assume `tcsh` on FreeBSD) and adjust accordingly. Either that or just launch bash in the script.

**Comment by euantorano** [comment]
It looks like part of the problem here is that `kochExe` has a null (`\0`) terminator in FreeBSD which causes it to be quoted even when in a safe path, so now I'm trying to figure out the cause for that...

**Comment by euantorano** [comment]
This is partially fixed by 16fc1de when there are no spaces in the path to Koch.
